ext/web/02_timers.js

```diff
@@ -48,14 +48,29 @@ function setImmediate(callback, ...args) {
  */
 function setTimeout(callback, timeout = 0, ...args) {
   checkThis(this);
+  const snapshot = core.AsyncContext.snapshot();
```

Comment:
Mirroring my comment from `deno_core` - is there a way to do this conditionally only if async context is actually in use? Some users might not care about it and it would be a shame to perf tax everyone.

Reply:
Answer here: https://github.com/denoland/deno_core/pull/804#discussion_r1663292304

Reply:
i will do lots of benchmarks when the v8 changes are available, though my initial hypothesis is that it will actually be the same or faster to just unconditionally snapshot/swap.